clients/src/main/java/org/apache/kafka/clients/producer/internals/IncompleteBatches.java

```diff
@@ -45,7 +46,7 @@ public void remove(ProducerBatch batch) {
         }
     }
 
-    public Iterable<ProducerBatch> copyAll() {
+    public List<ProducerBatch> copyAll() {
```

Comment:
q: why do we need to change it to type List?

Reply:
To make it testable, or easier to test, the `List` still could be get as an `Iterable`, in fact, that class did not have any test, this PR is adding the tests for it.